tools/check_installation.py

```diff
@@ -0,0 +1,85 @@
+"""
+Script for checking if all the test files are installed after building.
+
+Examples::
+
+    $ python check_installation.py install_directory_name
+
+        install_directory_name:
+            the relative path to the directory where SciPy is installed after
+            building and running `meson install`.
+
+Notes
+=====
+
+This script assumes that ``scipy`` and ``install_directory`` are present under
+the same parent folder.
+
+The script will stop on encountering the first missing file in the install dir,
+it will not give a full listing. This should be okay, because the script is
+meant for use in CI so it's not like many files will be missing at once.
+
+"""
+
+import os
+import glob
+import sys
+
+
+CUR_DIR = os.path.abspath(os.path.join(os.path.dirname(__file__)))
+ROOT_DIR = os.path.dirname(CUR_DIR)
+SCIPY_DIR = os.path.join(ROOT_DIR, 'scipy')
+
+
+# Files whose installation path will be different from original one
+changed_installed_path = {
+    'scipy/_build_utils/tests/test_scipy_version.py':
+        'scipy/_lib/tests/test_scipy_version.py'
+}
+
+
+def main(install_dir):
+    INSTALLED_DIR = os.path.join(ROOT_DIR, install_dir)
+    if not os.path.exists(INSTALLED_DIR):
+        raise ValueError(f"Provided install dir {INSTALLED_DIR} does not exist")
+
+    scipy_test_files = get_test_files(SCIPY_DIR)
+    installed_test_files = get_test_files(INSTALLED_DIR)
+
+    # Check test files detected in repo are installed
+    for test_file in scipy_test_files.keys():
+        if not test_file in installed_test_files.keys():
+            raise Exception("%s is not installed" % scipy_test_files[test_file])
+
+    # FIXME: this is the reverse test, but it does not seem possible to make
+    # this fail unless a file is installed twice (which is hard to do). If I
+    # move a file somewhere else, the check above already fails. So is this
+    # needed?
```

Comment:
Could you have a look at this @Smit-create? It doesn't quite do the right thing.

Reply:
Yeah, I got that. It will fail above itself if it is not installed on correct location or if it is missing. We can remove that. This can only be useful to make debugging faster so that we can match the missing file with improper installed file. But since we are raising an Exception, I don't  think this will be needed anymore.